### ISSUE TITLE
CreateInnerTransaction syntax is very verbose and not strict

### ISSUE THREAD

**Issue by ismael-elatifi** [algorandfoundation/puya#24]
First, congrats for your impressive work !
Reading the auction example, I was thinking that the syntax to send inner transaction is repeating many times `CreateInnerTransaction`.
It would be nice if there was a builtin more succint syntax avoiding this repetition.
Maybe something like a single fonction call by passing the multiple parameters of the transaction.

For example the current :  
```
CreateInnerTransaction.begin()
CreateInnerTransaction.set_type_enum(TransactionType.AssetTransfer)
CreateInnerTransaction.set_fee(UInt64(0))  # cover fee with outer txn
CreateInnerTransaction.set_asset_receiver(Global.current_application_address())
CreateInnerTransaction.set_xfer_asset(asset.asset_id)
CreateInnerTransaction.submit()
```
could be shorten to :  
```
InnerTransaction(
    type_enum=TransactionType.AssetTransfer, 
    fee=UInt64(0), 
    asset_receiver=Global.current_application_address(), 
    xfer_asset=asset.asset_id
).submit()
```

Maybe even better would be to expose only `InnerPaymentTransaction` and `InnerAssetTransferTransaction` constructors to make it explicit what parameters are required for `Payment` and for `AssetTransfer` (for example passing `xfer_asset` is forbiden for a `Payment`, this is what I mean by "strict").

**Comment by robdmoore** [comment]
Thanks for raising! Completely agree.

This one was definitely on our list already, right now it mirrors the specific TEAL op codes, but it's a no brainer to add more convenient methods like `PaymentTransaction(...)` etc.

**Comment by daniel-makerx** [comment]
Implemented in [0.6.0 ](https://github.com/algorandfoundation/puya/releases/tag/v0.6.0)
